solver.rb

```diff
@@ -0,0 +1,28 @@
+class Solver
```

Comment:
- [ ] From your commit history you added the class and methods before writing the tests, but you were supposed to write the tests before adding the class, as that is the purpose of the project. Kindly do that.